File: iotOnGolang/internal/model/SoftwareUpdateExecution.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// SoftwareUpdateExecution Declaration
//==============================================================
type SoftwareUpdateExecution struct {
    gorm.Model
     StartedAt                                                            time.Time
    CompletedAt                                                            time.Time
    CampaignId         *uint
    Campaign           *SoftwareUpdateCampaign `gorm:"foreignKey:CampaignId"`
    DeviceId         *uint
    Device           *IoTDevice `gorm:"foreignKey:DeviceId"`
    Status                      UpdateStatus

// parent associations as their child

}

File: iotOnDjango/models/SoftwareUpdateExecution.py

from django.db import models
from iotOnDjango.models.UpdateStatus import UpdateStatus

#======================================================================
# Class SoftwareUpdateExecution Declaration
#======================================================================
class SoftwareUpdateExecution (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	startedAt = models.CharField(max_length=64, null=True)
	completedAt = models.CharField(max_length=64, null=True)
	campaign = models.ForeignKey('SoftwareUpdateCampaign', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	device = models.ForeignKey('IoTDevice', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	status = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in UpdateStatus])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.startedAt
		str = str + self.completedAt
		str = str + self.status
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "SoftwareUpdateExecution";
    
	def objectType(self):
		return "SoftwareUpdateExecution";
